### DISCUSSION TITLE
How to correctly integrate WalletConnect 2.0 in Unity?

### DISCUSSION THREAD

**Discussion by Fatorin** [WalletConnect/walletconnect-monorepo#2422] [Developer Support]
I'm attempting to integrate WalletConnect 2.0 with Unity and encountering some issues. Here's my process so far:
1.Imported WalletConnect 2.0 into my Unity project using NuGetForUnity. It can't use "dotnet add" to import.
2.Modified the SignClientOptions and ConnectOptions to match the correct chain ID.
3.Converted the URI generated by WalletConnectSignClient into a QR code and displayed it on the screen.
4.Scanned the QR code with my phone and approved the connection.

A few issues I've run into are:
1.The official documentation mentions a 'SessionData' class that I'm unable to locate in the actual library.
2.After approving the connection on my phone, I'm not receiving any confirmation of successful connection in Unity.

Given these issues, I would greatly appreciate any guidance on the correct method to implement this.

Thank you in advance for your assistance!"

**Comment by finessevanes** [comment]
Hey - letting you know I've let our Unity team know. You should be expecting a message soon!

**Reply by finessevanes** [reply]
@Fatorin apologies for the delay in response. I've reached out to our Unity team. Appreciate your patience!

**Comment by ngmisl** [comment]
relevant question... there is no updated unpkg and no working unity plugin either

**Comment by iulianbroasca** [comment]
Hey,

@finessevanes Can you post it here or update the Unity page with the correct integration method, please?

Can you also give us some usage examples, maybe a Unity project with WalletConnectV2 in the project and some examples?

Thank you! ☺

**Reply by finessevanes** [reply]
Thank you for your suggestion! I've reached out to our Unity team for additional support.

**Comment by WoodsFiend** [comment]
Any update on this @finessevanes ? 2 days until V1 shutdown and still no guidance to upgrade.

**Reply by finessevanes** [reply]
Hey this is what you want to be using: https://docs.walletconnect.com/2.0/unity/sign/dapp-usage
